### DISCUSSION TITLE
0.51.0

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#7930] [General]
* Update Kernel to version 5.10.81.1
    * Update to upstream stable kernel release 5.10.81
    * Unify the kernel configurations by enabling missing options on ARM64
        * Enable non-arch specific ACPI options
        * Enable options related to device-mapper RAID
        * Enable Btrfs
        * Enable LZO and ZSTD compression
* Update WSLg to 1.0.30
    https://github.com/microsoft/wslg/releases/tag/v1.0.30
* Fix issue where WSLg TS DVC plugin was not correctly registered.
* Update MSRDC to version 1.2.2677
* Change path translation to treat `\\wsl.localhost\distro` and `\\wsl$\distro` prefix as case-insensitive [GH 7744]
* Add a specialized error message for `wsl.exe --mount` on ARM64
* 9p: Retry epoll_wait on EINTR instead of throwing [GH 7459]
* Add new `--web-download` to download distributions via http instead of using the store.  
* Add BOM support for .wslconfig
* Switch `wsl.exe --mount` and `wsl.exe --unmount` print error strings instead of integers.
* Introduce debug shell for debugging the root namespace.
    This change introduces new "debug shell" functionality to WSL. This will launch an interactive shell in the root Mariner namespace when the WSL2 VM launches. To enable the debug shell put the following entry in `%userprofile%\.wslconfig`
    ```
    [wsl2]
    debugShell=true
    ```

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/0.51.0'>0.51.0</a>.</em>

**Comment by torshind** [comment]
Hello there.

I'm honestly confused by how update deliveries are handled.

My understanding was that WSL as a windows module is sort of a stable branch, while the preview app is sort of a pre-release.

However, this is already the third app update that is not propagated to the windows module.
The current version installed by wsl --update (kernel 5.10.60.1 + WSLg 1.0.26) is rather stable, but I suffer from annoying CLion freezes and minor graphical anomalies, so I'd be happy to upgrade. However, I'm not sure if the preview version is stable enough.

In addition to this, the Windows store continues to not provide the version number, to either the store GUI or winget, which is really upsetting.

Could you please clarify?

**Reply by mangkoran** [reply]
In addition to that, the update still not available via `winget` nor Microsoft Store.

**Reply by benhillis** [reply]
I sorted out an issue with the store submission - the update (now with version 0.51.2) should be available in a day or so.

**Reply by mangkoran** [reply]
Is version 0.51.2 not released in Github?
